technical-docs/designs/repos-merge.md

```diff
@@ -0,0 +1,285 @@
+```
+
+#### Proposed Changes
+
+- Create `src` folder in `cluster` folder
+- Move all files and folders except `src` from `cluster` to `cluster\src`
+- Move `azure-pipelines.yml` file from the root to `cluster\build\pipelines`
+  directory and make sure all pipelines still works.
+- Move `docs` folder to `cluster` folder and make sure all doc links works.
+- Move `test` folder to `cluster` folder and make sure all tests and pipelines
+  still works.
+- Move `tools` directory to `cluster\build` folder and make sure existing
+  Bedrock release process still works.
+- [ ] Move following files form the `root` directory to `cluster` and make sure
+      `cluster` project can be opened in VS Code.
+  - LICENSE
+  - README.md
+  - .gitignore
+
+#### Testing Changes
+
+After making the above changes, need to make sure that the follwing tests pass.
+
+- Verify the environments integration tests continue to work in `azdo`
+- Verify all `docs` links work
+- Verify `bedrock` release provess work
+- Verify the project can be opened in VS Code from the `cluster` folder
+
+#### Bedrock repo after changes
+
+`Bedrock` repo would look like below after implementing the above changes.
+
+```
+├── .github
+├── gitops
+├── cluster
+    ├── build
+        ├── pipelines
+            ├── azure-pipelines.yml
+        ├── tools
+    ├── docs
+    ├── src
+    ├── test
+    ├── LICENSE
+    ├── README.md
+    ├── .gitignore
+
+```
+
+### Changes to SPK Repo
+
+Making following changes in existing SPK repo minimizes the number of changes
+after the merge and limits the testing scope to SPK functioanlity.
+
+#### Current SPK repo
+
+```
+├── .github
+├── azure-pipelines
+    ├── templates
+├── docs
+    ├── commands
+├── guides
+├── patches
+├── scripts
+├── src
+├── technical-docs
+    ├── designs
+├── tests
+├── tools
+├── typings
+    ├── ssh-url
+├── .editorconfig
+├── .env.example
+├── .gitignore
+├── CHANGELOG.md
+├── README.md
+├── azure-pipelines.yml
+├── jest.config.js
+├── package.json
+├── release-pipeline.yml
+├── smoke-test-pipeline.yml
+├── spk-config.yaml
+├── tsconfig.json
+├── tslint.json
+├── typings
+├── webpack.config.js
+├── .yarn.lock
+```
+
+#### Themes
+
+The main themes are related to organizing `docs` and `build` related artifacts
+in `spk` from the root to appropriate directories.
+
+1. Oragnize all pipeline yaml files, scripts, and tools that are related to
+   building and releasing binaries under `build` directory.
+
+2. Organize all docs that are related to using and contributing bedrock under
+   `docs` directory.
+
+#### Proposed Changes
+
+The majority of the proposed changes in `spk` repo are related to organizing
+`docs` and `build` folders.
+
+- Move following files from the `root` to `build\pipelines` folder and make sure
+  all pipelines continue to work.
+  - azure-pipelines.yml
+  - release-pipeline.yml
+  - smoke-test-pipeline.yml
+- Move following files from the `scripts` to `build\tools` folder and verify the
```

Comment:
I would rename `build\tools` to `build\publishing` to indicate that this files are related to the publishing of documentation. Just a thought